spec/lib/site_settings/validations_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+require 'site_settings/validations'
+
+describe SiteSettings::Validations do
+  subject { Class.new.include(described_class).new }
+
+  context "s3 buckets reusage" do
+    let(:error_message) { I18n.t("errors.site_settings.s3_bucket_reused") }
+
+    shared_examples "s3 bucket validation" do
+      def change_other_bucket_value(value)
+        SiteSetting.send("#{other_setting_name}=", value)
+      end
+
+      it "shouldn't raise an error when both buckets are blank" do
+        change_other_bucket_value("")
+        expect { validate("") }.to_not raise_error
+      end
+
+      it "shouldn't raise an error when only one bucket is set" do
+        change_other_bucket_value("")
+        expect { validate("my-awesome-bucket") }.to_not raise_error
```

Comment:
`to_not raise_error` [should be deprecated in a later version](https://github.com/rspec/rspec-expectations/pull/244) if I am not wrong. Anyway you can read more about the problem with it here -> https://github.com/rspec/rspec-expectations/issues/231

Reply:
I didn't read anything about deprecating `not_to raise_error` (without an error class). I think the more specific `expect { }.not_to raise_error(SpecificErrorClass)` doesn't even work anymore. I like the `to_not raise_error`, because it's more explicit to me as a reader, but I guess you are right and the "not raise an error" is already in the test name. Yeah, I'm going to remove it.

Reply:
Oops! I forgot to push that change before I merged. 😊 
Fixed in https://github.com/discourse/discourse/commit/2bdbca38012c4b46d8788a11369d968eb8abe885

Reply:
> I think the more specific expect { }.not_to raise_error(SpecificErrorClass) doesn't even work anymore

OO it was for a specific error. I've been holding this thought in my head that `not_to raise_error` is deprecated.